### PR TITLE
[Bugfix] Clean up some cruft in mamba.py

### PR DESCRIPTION
**Pull request by tlrmchlsmth**
Delete some unnecessary/useless code I noticed while reviewing #7478

### CODE REVIEW

**Review by DarkLight1337 — APPROVED** [review]

File: vllm/model_executor/models/mamba.py [hunk 1]
```diff
@@ -346,18 +338,8 @@ def forward(
...
-            "v_proj",
-        ],
-    }
-
     # LoRA specific attributes
     supported_lora_modules = [
-        "qkv_proj",
-        "o_proj",
         "embed_tokens",
         "lm_head",
```
Comment (mgoin) [inline vllm/model_executor/models/mamba.py:344]: I don't see these modules in the checkpoint either https://huggingface.co/state-spaces/mamba-130m-hf?show_file_info=model.safetensors

File: vllm/model_executor/models/mamba.py [hunk 2]
```diff
@@ -459,24 +441,15 @@ def sample(
     def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
         stacked_params_mapping = [
             # (param_name, shard_name, shard_id)
-            ("qkv_proj", "q_proj", "q"),
-            ("qkv_proj", "k_proj", "k"),
-            ("qkv_proj", "v_proj", "v"),
             ("gate_up_proj", "gate_proj", 0),
             ("gate_up_proj", "up_proj", 1),
```
Comment (mgoin) [inline vllm/model_executor/models/mamba.py:445]: I don't see these modules in the checkpoint either https://huggingface.co/state-spaces/mamba-130m-hf?show_file_info=model.safetensors

**Comment by tlrmchlsmth** [comment]
Ok, this was extremely boneheaded -- just removed an entire feed forwards section that I had copy-pasted from Jamba, that isn't present in the Mamba model.

File: vllm/model_executor/models/mamba.py [hunk 3]
```diff
@@ -332,40 +282,20 @@ def forward(
             current_ssm_state = ssm_state[i]
             current_conv_state = conv_state[i]
 
-            hidden_states, residual = layer(
+            hidden_states = layer(
                 positions=positions,
                 hidden_states=hidden_states,
                 attn_metadata=attn_metadata,
                 residual=residual,
```
Comment (mgoin) [inline vllm/model_executor/models/mamba.py:289]: Residual is still passed in here. I personally think it is fine to keep the previous structure of passing residual into rmsnorm, but up to you

**Review by mgoin — APPROVED** [review]
Now that is clean 😎

**Pull request merged**